cmd/medius/images/push.go

```diff
@@ -32,26 +28,9 @@ func NewPublishImagesCommand(options *common.Options) *cobra.Command {
 		Use:   "push",
 		Short: "Determine if containerdisks need an update and push an update to the target registry if needed",
 		Run: func(cmd *cobra.Command, args []string) {
-			errChan := make(chan error, options.PublishImagesOptions.Workers)
-			jobChan := make(chan api.Artifact, options.PublishImagesOptions.Workers)
-
-			wg := &sync.WaitGroup{}
-			wg.Add(options.PublishImagesOptions.Workers)
-			for x := 0; x < options.PublishImagesOptions.Workers; x++ {
-				go worker(wg, jobChan, options, errChan)
-			}
-
-			for i, desc := range common.Registry {
-				if options.Focus == "" && desc.SkipWhenNotFocused {
-					continue
-				}
-				if options.Focus != "" && options.Focus != desc.Artifact.Metadata().Describe() {
-					continue
-				}
-				jobChan <- common.Registry[i].Artifact
-			}
-			close(jobChan)
-
+			wg, errChan := spawnWorkers(options.PublishImagesOptions.Workers, options.Focus, func(a api.Artifact) error {
```

Comment:
I find this pattern unusual, where a function returns a `WaitGroup` and then immediately waits on it. 
Can you move the `wg.Wait()` inside the function and make it a blocking function?